File: ComplexFoodProjectStructureBE/Domain/Models/Roles/Buyer.cs
using Domain.Models.Ordering;
using Domain.Models.Shopping;
using Domain.ValueObjects;

namespace Domain.Models.Roles;

public class Buyer
{
    public int Id { get; set; }
    public Email Email { get; set; }
    public Password Password { get; set; }
    public Name FirstName { get; set; } 
    public Name LastName { get; set; }
    public PhoneNumber PhoneNumber { get; set; }
    public Gender Gender { get; set; }
    public bool Confirmed { get; set; }
    public Balance Balance { get; set; }

    //one to many
    public ICollection<Coupon> Coupons { get; set; } = new List<Coupon>();
   

}

File: Frontend/04.Typescript Introducing/TypescriptAssignment/Domain/Roles/Buyer.ts
import { IBuyer } from '../../Application/Contracts/IBuyer';

export class Buyer implements IBuyer {
  Email: string;
  Password: string;
  FirstName: string;
  LastName: string;
  PhoneNumber: string;
  Gender: string;
  Confirmed: boolean;
  Balance: Float32Array;
}
